### PR TITLE
CYS - AI Offline flow

### LINKED ISSUE #41623: [CYS] Make sure patterns with default content are displayed in case AI is unavailable.
Make sure the user is still sent to the Assembler page with the default patterns content if AI is temporarily unavailable.

### 1. Identifying if AI is available

Make a GET request to `https://status.openai.com/api/v2/status.json` to check its status. If the returned indicator is set as "major" or "critical," we can consider AI is down. For more details about the response format for this API, please refer to the documentation on https://status.openai.com/api/v2#status 

### 2. On failure, reset all patterns content

If AI is down, make a DELETE request to the `ai/patterns` endpoint on Store API to ensure all patterns are reset to the default content on the client side whenever AI is unavailable. 

PS: the DELETE request for `ai/patterns` doesn't exist yet - it is planned to be implemented by the Store Editing team soon. Relevant issue is available here: https://github.com/woocommerce/woocommerce-blocks/issues/11873

### 3. On failure, direct the users to the Assembler. Create a modal as demonstrated in the following video

Video with the complete walkthrough:

https://github.com/woocommerce/woocommerce/assets/15730971/4e40b313-16bb-425a-a129-ba4b0fbe1c3d

Screenshots of the modals displayed in the video:

| Modal 1 | Modal 2 | Modal 3 |
| ------ | ----- | ----- |
| Screenshot 2023-11-21 at 21 45 28 Mrk6SERPZ4KrFHSjM0a8TK-fi-7573_46973 | Screenshot 2023-11-21 at 21 46 51 Mrk6SERPZ4KrFHSjM0a8TK-fi-7573_112782 | Screenshot 2023-11-21 at 21 47 06 Mrk6SERPZ4KrFHSjM0a8TK-fi-7573_112760 |

### PR DESCRIPTION
**Pull request by moon0326**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #41623  

This pull request addresses situations in which the AI is offline on the intro page or when making a request to one of the AI endpoints.

### How to test the changes in this Pull Request:

1. Create a new site either via local environment or JN. Make sure to install WooCommerce Beta Tester as well.
2. Connect your site with Jetpack
3. Enable `customize-store` feature flag in `Tools -> WCA Test Helper -> Features`
4. Install and activate  the nightly build of WooCommerce Blocks https://github.com/woocommerce/woocommerce-blocks/releases/download/nightly/woocommerce-blocks-trunk-nightly.zip
5. Go to `WooCommerce -> Home` and click on `Customize your store` task.

#### AI is online when the user lands on the intro page

1. Click on `Design with AI` button and proceed to the assembler hub.
2. Confirm call to `/wc/private/ai/patterns` and response is `{"ai_content_generated":true}`.

#### AI is offline when the user lands on the intro page

1.  Go to `WooCommerce -> Home` and click on `Customize your store` task.
2. Open your browser's inspector and use Chrome's content override [feature](https://developer.chrome.com/docs/devtools/overrides/) to change the content of https://status.openai.com/api/v2/status.json call. Change `status.indicator` to `major`. 
3. Perform a hard refresh
4. Click on `Design with AI` button and proceed to the assembler hub.
5.  Confirm call to `/wc/private/ai/patterns` and response is `removed: true` in the network tab (inspector)
6. The Welcome modal should have the updated copy Mrk6SERPZ4KrFHSjM0a8TK-fi-7573%3A50973

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [x] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

Tweaks for the CYS when the A.I is offline.

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-admin/client/customize-store/intro/services.ts [hunk 1]
```diff
@@ -8,6 +8,25 @@ import { resolveSelect } from '@wordpress/data';
...
+export const fetchAiStatus = () => async (): Promise< aiStatusResponse > => {
+	const response = await fetch(
+		'https://status.openai.com/api/v2/status.json'
+	);
+	const data = await response.json();
+	return data;
+};
+
+// Reset all pattern contents.
+// To be implemented in a future PR.
```
Comment (nefeline) [inline plugins/woocommerce-admin/client/customize-store/intro/services.ts]: Mentioned on the issue, but sharing here as well just as a heads up: the endpoint for resetting the patterns was already created on WooCommerce Blocks: https://github.com/woocommerce/woocommerce-blocks/pull/11890

**Review by nefeline — COMMENTED** [review]
Great work here @moon0326 !

Confirmed the modals were updated with the correct content:

Screenshot 2023-11-28 at 10 18 11
Screenshot 2023-11-28 at 10 17 53
Screenshot 2023-11-28 at 10 17 30

The only problem I noticed is that we are not displaying any patterns at all when the loading is finished: when the assembler is displayed, I'm being presented with the default TT3 homepage instead:




https://github.com/woocommerce/woocommerce/assets/15730971/75c563b9-5353-4f39-a5e4-53cd2fb202a5

File: plugins/woocommerce-admin/client/customize-store/intro/services.ts (hunk 1 above)
Comment (moon0326) [reply plugins/woocommerce-admin/client/customize-store/intro/services.ts]: Thanks for the reminder @nefeline 

Updated in https://github.com/woocommerce/woocommerce/pull/41656/commits/98ac938aadeddb7f74ed23123cadb273f0b8e046

**Comment by moon0326** [comment]
Hey @chihsuan Could you please take a look at https://github.com/woocommerce/woocommerce/pull/41656#pullrequestreview-1752305381 comment when you get a chance 🙏 ? I'm not 100% familiar with how the patterns get populated when AI is not used.

**Comment by moon0326** [comment]
@nefeline Given it's almost the end of the week, I've updated the code and test instructions to reflect changes from pdnLyh-4NL-p2 (I understand we still have unanswered questions, but the changes were quick to make.). I'll wait for the copy change confirmation before making other changes.

**Comment by nefeline** [comment]
> @nefeline Given it's almost the end of the week, I've updated the code and test instructions to reflect changes from pdnLyh-4NL-p2 (I understand we still have unanswered questions, but the changes were quick to make.). I'll wait for the copy change confirmation before making other changes.

Hi @moon0326, could you please clarify what changes were made here? Can you please confirm that we are still aligned with the scope as described on https://github.com/woocommerce/woocommerce/issues/41623? 

FYI: Based on the outcome of that discussion on P2, the scope of the original issue should remain the same. pdnLyh-4NL-p2#comment-3498

**Comment by moon0326** [comment]
> > @nefeline Given it's almost the end of the week, I've updated the code and test instructions to reflect changes from pdnLyh-4NL-p2 (I understand we still have unanswered questions, but the changes were quick to make.). I'll wait for the copy change confirmation before making other changes.
> 
> Hi @moon0326, could you please clarify what changes were made here? Can you please confirm that we are still aligned with the scope as described on #41623?
> 
> FYI: Based on the outcome of that discussion on P2, the scope of the original issue should remain the same. pdnLyh-4NL-p2#comment-3498

I've made the following changes:

AI offline
1. Render the intro page without AI buttons (screenshot below) when the AI is offline.
https://github.com/woocommerce/woocommerce/pull/41656/commits/9b0e4c9821cb0c308874fb7b1c428dae64218fce

AI online
1. When one of the AI endpoints fails after users submit the AI questions, we consider the AI offline and use the updated copies and patterns.
https://github.com/woocommerce/woocommerce/pull/41656/commits/52435af2116ba1b2c83f850675d91b4dea7c82dc


> FYI: Based on the outcome of that discussion on P2, the scope of the original issue should remain the same. pdnLyh-4NL-p2#comment-3498

TBH, It's still a bit confusing to me because the issue doesn't actually describe what's on the Figma file.

1. It does not describe the intro page with the theme selection when the AI is offline.
2. It still has a video that excludes business info and look and feel pages.
3. While it only mentions checking OpenAI status at the start, we actually need to monitor failures of AI endpoints from the /wc/private/ai/* endpoints as well.

**Comment by nefeline** [comment]
Hi @moon0326 ! Thanks for sharing the details of your recent changes. Regarding:

> It does not describe the intro page with the theme selection when the AI is offline.

We don't want to implement this change: what we aim for here is what is described on the issue: "On failure, direct the users to the Assembler. Create a modal as demonstrated in the following video". The theme selection when the AI is offline option was discarded and won't be implemented: we are proceeding with the most robust long-term solution, which is directing users to the Assembler with the patterns with the default content, as you had already implemented initially here. I tested your PR, and I see it is now deviating from the issue description, directing users to the theme selection instead: let's revert this change and roll back to the original scope, please.

> It still has a video that excludes business info and look and feel pages.

That is the expected based on the feedback and confirmation we received both from @jarekmorawski  and @verofasulo on the issue opened for this task: let's stick with their recommendation, please.

> While it only mentions checking OpenAI status at the start, we actually need to monitor failures of AI endpoints from the /wc/private/ai/* endpoints as well.

Handling errors from the patterns endpoint is not part of the scope of this issue, only the AI offline state: the number of retries and what happens when they fail will be handled as part of a separate task.

In you have any additional questions/concerns, please let me know!

**Comment by chihsuan** [comment]
> Hey @chihsuan Could you please take a look at [#41656 (review)](https://github.com/woocommerce/woocommerce/pull/41656#pullrequestreview-1752305381) comment when you get a chance 🙏 ? I'm not 100% familiar with how the patterns get populated when AI is not used.

@moon0326 Sorry, I missed that. The home template can be updated by calling `assembleSite` service.
 https://github.com/woocommerce/woocommerce/blob/b6657403f83e5e7e058542d5100a6edac1a2d804/plugins/woocommerce-admin/client/customize-store/design-with-ai/services.ts#L403-L440

**Review by nefeline — APPROVED** [review]
Thanks for your great work here @moon0326 and @chihsuan! Tested and confirmed things are working as expected:


https://github.com/woocommerce/woocommerce/assets/15730971/28c89452-22f9-47e5-ad88-75b97026d19c

**Pull request merged**